fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -425,6 +429,74 @@ void testQueryWithStopWords() {
         }
     }
 
+    /**
+     * Test that lucene doesn't break whene there's unicode in the text.
+     * <p>
+     *     This test is not intending to test the behavior of querying with unicode, just that having it doesn't break
+     *     other queries.
+     * </p>
+     */
+    @Test
+    void testQueryWithUnicode() {
+        List<StringBuilder> allBuilders = new ArrayList<>();
+        final String beginningWord = "banana";
+        final String endWord = "catamaran";
+        BiConsumer<StringBuilder, String> addWord = (stringBuilder, word) -> stringBuilder.append(" ").append(word).append(" ");
+        StringBuilder sb = new StringBuilder();
+        addWord.accept(sb, beginningWord);
+        allBuilders.add(sb);
+        int skipped = 0;
+        for (int i = 0; i < Character.MAX_CODE_POINT; i++) {
+            if (Character.isSupplementaryCodePoint(i)) {
+                sb.append(Character.highSurrogate(i));
+                sb.append(Character.lowSurrogate(i));
+            } else if (Character.isDefined(i) && Character.getType(i) != Character.PRIVATE_USE
+                    && !Character.isSurrogate((char) i)) {
+                sb.append((char) i);
+            } else {
+                skipped++;
+            }
+            if (sb.length() > 50_000) {
+                addWord.accept(sb, endWord);
+                sb = new StringBuilder();
+                addWord.accept(sb, beginningWord);
+                allBuilders.add(sb);
+            }
+        }
+        addWord.accept(allBuilders.get(allBuilders.size() - 1), endWord);
+        List<String> allTexts = allBuilders.stream().map(StringBuilder::toString).collect(Collectors.toList());
+        LOGGER.debug("Skipped: " + skipped + "/" + Character.MAX_CODE_POINT);
+        LOGGER.debug("AllTexts (" + allTexts.size() + "): " + allTexts.stream().map(text -> String.valueOf(text.length())).collect(Collectors.joining(", ")));
+        long id = 1L;
+        for (final String text : allTexts) {
+            try (FDBRecordContext context = openContext()) {
+                openRecordStore(context);
+                TestRecordsTextProto.SimpleDocument document1 = TestRecordsTextProto.SimpleDocument.newBuilder()
+                        .setDocId(id++)
+                        .setGroup(1)
+                        .setText(text)
+                        .build();
+                recordStore.saveRecord(document1);
+                TestRecordsTextProto.SimpleDocument document2 = TestRecordsTextProto.SimpleDocument.newBuilder()
+                        .setDocId(id++)
+                        .setGroup(1)
+                        .setText(beginningWord + " Good morning the Mr Tian, would you like a " + endWord)
+                        .build();
+                recordStore.saveRecord(document2);
```

Comment:
What's the purpose of `document2`?

Reply:
I think it was just a side effect of copying the test, and I didn't clean it up. It's removed now.